lib/logstash/filters/mutate.rb

```diff
@@ -288,6 +291,17 @@ def convert_float(value)
     return value.to_f
   end # def convert_float
 
+  def convert_boolean(value)
+    if value == "true"
+      return true
+    elsif value == "false"
+      return false
+    else
+      @logger.warn("Failed to convert #{value} into boolean. Acceptable values are \"true\" and \"false\".")
+      return value
```

Comment:
when a string is neither false nor true, `convert_boolean` will print a debug, and return the original string.
This means that the calling parent, `convert`, will be a successful noop, right? Which means that actions like `add_field` and `remove_tag` will work, and shouldn't.
This method's `else` block should call next, like [gsub](https://github.com/untergeek/logstash-filter-mutate/blob/feature/20/lib/logstash/filters/mutate.rb#L326), what do you think?

This can be encapsulated in a test: configure the plugin to convert boolean and add_field "potatoes" => 1, send an event and check that a non boolean string should not have the potatoes field